### ISSUE TITLE
Implement missing client side stats

### ISSUE THREAD

**Issue by gibsong** [ehcache/ehcache3#1600]
onHeapStore
1. occupiedMemory

offheapStore
1. maxMappings

diskStore
1. maxMappings

clusteredStore
1. mappings
2. maxMappings
3. allocatedMemory
4. occupiedMemory

**Comment by fossf** [comment]
Can you validate if this is still an issue?

**Comment by mathieucarbou** [comment]
@fossf @chrisdennis @GaryWKeim :

we have:

- OnHeap:OccupiedByteSize
- OffHeap:MappingCount
- Disk:MappingCount

For the others:

- We do not have access to any max mapping count in Ehcache and I do not think it is relevant to have it

- For the clustered stats Ehcache is a client. Clustered data is hold onto servers, and this is where we get the stats. But they are not exposed through Ehcache at the moment, and on the servers we don't have them like described in the issue. They are on different form (i.e. `RestartStore:TotalUsage`, `Store:DataSize`, `OffHeapResource:AllocatedMemory`, `Pool:AllocatedSize`).

So depending on what is the goal, the answer to the question is yes or no.

My take is that we can close it because the way the data is stored on the servers it does not match the stats names we have in the client. 
And I am not sure it is right to expose these server stats from Ehcache. 

They could be useful though. I.e., as an open-source user, I would like to know the amount of memory taken by my Ehcache server entities...

So in conclusion, there are no "missing client stats" like the issue states, but instead we lack in Ehcache a way for the statistic service to report the usage of the Ehcache server entities on server (clustered layer). This could be useful because the OSS server-side part is quite a black box...

**Comment by chrisdennis** [comment]
Refactoring has made this issue obsolete.
